drift_dev/lib/src/analyzer/sql_queries/type_mapping.dart

```diff
@@ -120,14 +120,17 @@ class TypeMapper {
   ///    a Dart placeholder, its indexed is LOWER than that element. This means
   ///    that elements can be expanded into multiple variables without breaking
   ///    variables that appear after them.
-  List<FoundElement> extractElements(AnalysisContext ctx,
+  List<FoundElement> extractElements(AnalysisContext ctx, AstNode root,
       {RequiredVariables required = RequiredVariables.empty}) {
     // this contains variable references. For instance, SELECT :a = :a would
     // contain two entries, both referring to the same variable. To do that,
     // we use the fact that each variable has a unique index.
-    final variables = ctx.root.allDescendants.whereType<Variable>().toList();
+    final collector = _VariableCollector();
+    root.accept(collector, null);
+
+    final variables = collector.result;
     final placeholders =
-        ctx.root.allDescendants.whereType<DartPlaceholder>().toList();
+        root.allDescendants.whereType<DartPlaceholder>().toList();
```

Comment:
Because nested queries could contain variables that are not required by the main query, it is no longer possible to just use `root.allDescendants` to find all variables. Also all variables in subqueries are indexed separately .

Reply:
But I think for placeholder this should still be fine, idk how they work exactly.

Reply:
Placeholders are things like `ORDER BY $order`, which generates an `OrderBy` as a Dart parameter.

Thinking about queries like `SELECT ..., LIST (SELECT * FROM ... ORDER BY $order)`, wouldn't that be a problem then?